step/step_test.go

```diff
@@ -238,32 +238,37 @@ func defaultResult() Result {
 		XcresultPath:             "XcresultPath",
 		XcodebuildBuildLog:       "XcodebuildBuildLog",
 		XcodebuildTestLog:        "XcodebuildTestLog",
-		SimulatorDiagnosticsPath: "SimulatorDiagnosticsPath",
+		SimulatorDiagnosticsPath: "/testpath/SimulatorDiagnosticsPath",
 	}
 }
 
-func createStepAndMocks(envValues map[string]string) (XcodeTestRunner, testingMocks) {
-	envRepository := new(mocks.Repository)
-	call := envRepository.On("Get", mock.Anything)
-	call.RunFn = func(arguments mock.Arguments) {
-		key := arguments[0].(string)
-		value := envValues[key]
-		call.ReturnArguments = mock.Arguments{value, nil}
+func createStepAndMocks(t *testing.T, envValues map[string]string) (XcodeTestRunner, testingMocks) {
+	envRepository := mocks.NewRepository(t)
+
+	if envValues != nil {
+		call := envRepository.On("Get", mock.Anything)
+		call.RunFn = func(arguments mock.Arguments) {
+			key := arguments[0].(string)
+			value := envValues[key]
+			call.ReturnArguments = mock.Arguments{value, nil}
+		}
 	}
 
 	inputParser := stepconf.NewInputParser(envRepository)
-	installer := new(mocks.Installer)
-	xcodebuilder := new(mocks.Xcodebuild)
-	simulatorManager := new(mocks.Manager)
-	cache := new(mocks.SwiftPackageCache)
-	outputExporter := new(mocks.Exporter)
-	pathModifier := new(mocks.PathModifier)
-	pathProvider := new(mocks.PathProvider)
-
-	step := NewXcodeTestRunner(inputParser, log.NewLogger(), installer, xcodebuilder, simulatorManager, cache, outputExporter, pathModifier, pathProvider)
+	installer := mocks.NewInstaller(t)
```

Comment:
Using the mock constructors generated by mockery. These are cool as they call AssertExpectations automatically on test cleanup, so you  can not forget adding it.